infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutContextData.kt

```diff
@@ -0,0 +1,280 @@
+package fi.fta.geoviite.infra.tracklayout
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import fi.fta.geoviite.infra.common.*
+import fi.fta.geoviite.infra.common.DataType.STORED
+import fi.fta.geoviite.infra.common.DataType.TEMP
+import fi.fta.geoviite.infra.logging.Loggable
+
+// TODO: GVT-2426 This enum is an outdated concept that won't serve well with designs. It's only used by the front. Can we refactor it out?
```

Comment:
Tämän poistaminen aiheuttaa aika paljon jumppaa UI-puolella. Katson sitä erikseen tämän pullarin jälkeen.

Reply:
Mitä jos tästä unohtaisi kokonaan tuon MAIN-/DESIGN-erottelun ja jättäisi tämän vaan wanhanmalliseksi `OFFICIAL`-/`NEW_DRAFT`-/`EDITED_DRAFT`-enumiksi? Tällöin siis enumin skooppi muuttuisi käytännössä vain branchin sisäiseksi. Jos tämä on käytössä puhtaasti vain frontilla, niin näen että siellä valtaosassa tapauksista (ellei jopa ihan kaikissa tapauksissa) meitä kiinnostaa nimenomaan onko se muuttunut branchin sisällä, ja jos nuo on MAIN-/DESIGN- puolet ovat omat arvonsa, niin se johtaa lähinnä siihen että tarkasteltavat jutut iffien ehdoissa ym. pitää tuplata

Reply:
Teen tämän suhteen uuden kierroksen tämän branchin päälle, sen pohjalta mitä aamulla juteliin.